sdk/servicebus/service-bus/src/connectionContext.ts

```diff
@@ -125,24 +125,7 @@ export namespace ConnectionContext {
             "clients. We should reconnect.",
           connectionContext.connection.id
         );
-        await delay(Constants.connectionReconnectDelay);
-        // reconnect clients if any
-        for (const id of Object.keys(connectionContext.clientContexts)) {
-          const clientContext = connectionContext.clientContexts[id];
-          log.error(
-            "[%s] calling detached on client '%s'.",
-            connectionContext.connection.id,
-            clientContext.clientId
-          );
-          clientContext.onDetached(connectionError || contextError).catch((err) => {
-            log.error(
-              "[%s] An error occurred while reconnecting the sender '%s': %O.",
-              connectionContext.connection.id,
-              clientContext.clientId,
-              err
-            );
-          });
-        }
+        await reconnect(connectionContext, connectionError || contextError);
```

Comment:
`reconnect()` internally calls `cleanConnectionContext()`, but a few lines above there are some more clean up tasks. 
- removeAllSessions(): Do we need this anymore now that we are not re-using the connection object?
- cbsSession.close(): This is already being done inside `cleanConnectionContext()`. But, I would suggest doing a git blame and figure out why this was added. If this was added due to issues coming up due to us re-using the same connection object, then we don't need to do this anymore? 
- closing of the management client: Same as above

Reply:
Ah, your intuition appears to be right:
https://github.com/Azure/azure-sdk-for-js/pull/1450 (event-hubs but has more context than the service-bus PR)
The removeAllSessions and cbsSession.close calls were to prevent sessions from being brought back up by rhea, and to prevent auth issues when doing so.

I don't think these are needed anymore.

Reply:
If `connection.close()` fails which results in `cleanConnectionContext` failing, then having the above would be a good back up I think.

For Track 2, we can look into moving these directly inside `cleanConnectionContext`